### DISCUSSION TITLE
Hello. Could anyone help me understand how the LIKELY() and the false work in this marco? Thanks a lot.

### DISCUSSION THREAD

**Discussion by Yang-Unreal** [Sixze/ALS-Refactored#225] [Q&A]
![1673830457432](https://user-images.githubusercontent.com/65805377/212588891-93db88af-e388-4816-b4bc-ba0709edde56.png)

**Comment by ameaninglessname** [comment]
For the former part of the question:
[likely — unlikely directives
Assisting the compiler in optimizing if conditions](https://medium.com/software-design/likely-unlikely-directives-802c09bd5232)
